HazelcastProxyFactoryTest testProxyHazelcastInstanceClasses_ofSameVersion_areSame

- Fails on `Hazelcast-4.master-IbmJDK8-fips-nightly`, also `Hazelcast-4.master-nightly`
- Fails on [Build #162 (May 19, 2020 9:11:36 AM)](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-4.master-IbmJDK8-fips-nightly/162/testReport/com.hazelcast.test.starter.test/HazelcastProxyFactoryTest/testProxyHazelcastInstanceClasses_ofSameVersion_areSame/)
- Error
```
java.lang.reflect.InvocationTargetException
```
- Stacktrace
```
Caused by: java.lang.NoClassDefFoundError: com/hazelcast/internal/serialization/impl/SampleIdentifiedDataSerializable
	at com.hazelcast.internal.serialization.impl.TestSerializerHook.getSerializationType(TestSerializerHook.java:37)
	at com.hazelcast.internal.serialization.impl.SerializerHookLoader.load(SerializerHookLoader.java:63)
	at com.hazelcast.internal.serialization.impl.SerializerHookLoader.<init>(SerializerHookLoader.java:55)
	at com.hazelcast.internal.serialization.impl.DefaultSerializationServiceBuilder.registerSerializerHooks(DefaultSerializationServiceBuilder.java:310)
	at com.hazelcast.internal.serialization.impl.DefaultSerializationServiceBuilder.build(DefaultSerializationServiceBuilder.java:238)
	at com.hazelcast.internal.serialization.impl.DefaultSerializationServiceBuilder.build(DefaultSerializationServiceBuilder.java:55)
	at com.hazelcast.instance.impl.DefaultNodeExtension.createSerializationService(DefaultNodeExtension.java:252)
	at com.hazelcast.instance.impl.Node.<init>(Node.java:247)
	at com.hazelcast.instance.impl.HazelcastInstanceImpl.createNode(HazelcastInstanceImpl.java:147)
	at com.hazelcast.instance.impl.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:116)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:241)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:220)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:158)
	at com.hazelcast.core.Hazelcast.newHazelcastInstance(Hazelcast.java:57)
```
- Possible commit (commit: 15f8670) 

kindly check

--- a/hazelcast/src/test/java/com/hazelcast/test/TestLoggerFactory.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/TestLoggerFactory.java
@@ -16,7 +16,6 @@
 
 package com.hazelcast.test;
 
-import com.hazelcast.internal.util.Timer;
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.logging.Log4j2Factory;
 import com.hazelcast.logging.LogEvent;
@@ -118,21 +117,21 @@ public class TestLoggerFactory extends LoggerFactorySupport {
 
         @Override
         public void finest(String message) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.finest(message);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void finest(Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.finest(thrown);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void finest(String message, Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.finest(message, thrown);
             logOnSlowLogging(startNanos);
         }
@@ -144,21 +143,21 @@ public class TestLoggerFactory extends LoggerFactorySupport {
 
         @Override
         public void fine(String message) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.fine(message);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void fine(Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.fine(thrown);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void fine(String message, Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.fine(message, thrown);
             logOnSlowLogging(startNanos);
         }
@@ -170,21 +169,21 @@ public class TestLoggerFactory extends LoggerFactorySupport {
 
         @Override
         public void info(String message) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.info(message);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void info(String message, Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.info(message, thrown);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void info(Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.info(thrown);
             logOnSlowLogging(startNanos);
         }
@@ -196,21 +195,21 @@ public class TestLoggerFactory extends LoggerFactorySupport {
 
         @Override
         public void warning(String message) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.warning(message);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void warning(Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.warning(thrown);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void warning(String message, Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.warning(message, thrown);
             logOnSlowLogging(startNanos);
         }
@@ -222,21 +221,21 @@ public class TestLoggerFactory extends LoggerFactorySupport {
 
         @Override
         public void severe(String message) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.severe(message);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void severe(Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.severe(thrown);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void severe(String message, Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.severe(message, thrown);
             logOnSlowLogging(startNanos);
         }
@@ -248,21 +247,21 @@ public class TestLoggerFactory extends LoggerFactorySupport {
 
         @Override
         public void log(Level level, String message) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.log(level, message);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void log(Level level, String message, Throwable thrown) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.log(level, message, thrown);
             logOnSlowLogging(startNanos);
         }
 
         @Override
         public void log(LogEvent logEvent) {
-            long startNanos = Timer.nanos();
+            long startNanos = System.nanoTime();
             delegate.log(logEvent);
             logOnSlowLogging(startNanos);
         }
@@ -278,7 +277,7 @@ public class TestLoggerFactory extends LoggerFactorySupport {
         }
 
         private void logOnSlowLogging(long startTimeNanos) {
-            long durationNanos = Timer.nanosElapsed(startTimeNanos);
+            long durationNanos = System.nanoTime() - startTimeNanos;
             if (durationNanos > WARNING_THRESHOLD_NANOS) {
                 long durationMillis = NANOSECONDS.toMillis(durationNanos);
                 delegate.warning("Logging took " + durationMillis + " ms.");

--- a/hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastAPIDelegatingClassloader.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastAPIDelegatingClassloader.java
@@ -57,6 +57,7 @@ public class HazelcastAPIDelegatingClassloader extends URLClassLoader {
         Set<String> alwaysDelegateWhiteList = new HashSet<>();
         alwaysDelegateWhiteList.add("com.hazelcast.test.starter.ProxyInvocationHandler");
         alwaysDelegateWhiteList.add("com.hazelcast.test.starter.HazelcastAPIDelegatingClassloader");
+        alwaysDelegateWhiteList.add("com.hazelcast.internal.serialization.impl.SampleIdentifiedDataSerializable");
         DELEGATION_WHITE_LIST = Collections.unmodifiableSet(alwaysDelegateWhiteList);
     }
 

